src/ai/defaultai.cc

```diff
@@ -545,7 +545,9 @@ void DefaultAI::think() {
 			break;
 		case SchedulerTaskId::kDiplomacy:
 			diplomacy_actions(gametime);
-			set_taskpool_task_time(gametime + kDiplomacyInterval, SchedulerTaskId::kDiplomacy);
+			set_taskpool_task_time(
+			   gametime + kDiplomacyInterval + Duration(RNG::static_rand(30) * 1000),
```

Comment:
I believe this is to decouple the Diplomacy of each player from the other players correct?

Reply:
Exactly.

Reply:
It was annoying that the actions graph was always unreadabe because many accepts/denies were overlapping with the request. :) But also all checks depended on player ordering, which may sometimes favour one player over the other.

BTW, speaking of the graphs, should we include the script in `utils`?